### PR TITLE
Add restore_to_point_in_time variable

### PR DESCRIPTION
**Pull request by joe-niland**
## what

* Allow specifying the required configuration to create a database instance via the point-in-time-recovery process
* Update variable description for `snapshot_identifier`

## why

* Support an alternative creation/DR scenario

## references

None

### CODE REVIEW

File: docs/terraform.md [hunk 1]
```diff
@@ -10,7 +10,7 @@
 
 | Name | Version |
 |------|---------|
-| <a name="provider_aws"></a> [aws](#provider\_aws) | >= 5.0 |
+| <a name="provider_aws"></a> [aws](#provider\_aws) | 5.25.0 |
```
Comment (Gowiem) [inline docs/terraform.md:13]: This is an annoying bug with terraform-docs. You may want to update manually and push as this will fail tests.

**Review by Gowiem — CHANGES_REQUESTED** [review]
Going to fail tests so change the readme provider issue and we'll get this shipped! 👍

File: docs/terraform.md (hunk 1 above)
Comment (joe-niland) [reply docs/terraform.md]: Thanks @Gowiem! That fixed it.

**Review by Gowiem — APPROVED** [review]
:shipit:

**Pull request merged**